codegen/cmd/injection-gen/generators/packages.go

```diff
@@ -188,22 +188,27 @@ func MustParseClientGenTags(lines []string) Tags {
 	return ret
 }
 
-func extractReconcilerClassTag(t *types.Type) (string, bool) {
+func extractCommentTags(t *types.Type) map[string]map[string]string {
 	comments := append(append([]string{}, t.SecondClosestCommentLines...), t.CommentLines...)
-	values := types.ExtractCommentTags("+", comments)["genreconciler:class"]
-	for _, v := range values {
-		if len(v) == 0 {
-			continue
-		}
-		return v, true
+	return ExtractCommentTags("+", comments)
+}
+
+func extractReconcilerClassTag(tags map[string]map[string]string) (classname string, has bool) {
+	vals, has := tags["genreconciler"]
+	if !has {
+		return
 	}
-	return "", false
+	classname, _ = vals["class"]
+	return
 }
 
-func isNonNamespaced(t *types.Type) bool {
-	comments := append(append([]string{}, t.SecondClosestCommentLines...), t.CommentLines...)
-	_, nonNamespaced := types.ExtractCommentTags("+", comments)["genclient:nonNamespaced"]
-	return nonNamespaced
+func isNonNamespaced(tags map[string]map[string]string) bool {
+	vals, has := tags["genclient"]
+	if !has {
+		return false
+	}
+	_, has = vals["nonNamespaced"]
+	return has
```

Comment:
```suggestion
	return has && vals["nonNamespaced"]
```

Reply:
Looks the same.

Reply:
This suggestion does not compile. vals["nonNamespaced"] is not boolean.

Reply:
hm, for reason I thought it's a map to `bool`...